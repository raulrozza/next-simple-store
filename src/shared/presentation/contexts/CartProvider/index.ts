export { default as useAddToCart } from './hooks/useAddToCart';
