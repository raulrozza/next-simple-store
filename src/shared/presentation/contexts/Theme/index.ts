export { default as useTheme } from './hooks/useTheme';
export { default as Theme } from './implementations/StyledComponentTheme';
