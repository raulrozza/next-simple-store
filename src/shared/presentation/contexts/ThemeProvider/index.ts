export { default as useTheme } from './hooks/useTheme';
export { default as ThemeProvider } from './implementations/StyledComponentTheme';
