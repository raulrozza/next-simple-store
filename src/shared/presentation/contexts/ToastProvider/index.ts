export { useToastProvider } from './context';
export { default as ToastProvider } from './implementations/ReactToastifyProvider';
