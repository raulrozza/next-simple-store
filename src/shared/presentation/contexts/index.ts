export * from './CartProvider';
export * from './ThemeProvider';
