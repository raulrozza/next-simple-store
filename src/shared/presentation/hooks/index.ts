import { trpc } from '@/shared/infra/http/trpc';

const { useQuery, useMutation } = trpc;

export { useQuery, useMutation };
