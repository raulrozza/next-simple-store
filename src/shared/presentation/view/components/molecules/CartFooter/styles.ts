import styled, { css } from 'styled-components';

export const Footer = styled.footer`
    display: flex;
    flex-direction: column;
    align-items: flex-start;

    ${({ theme }) => css`
        color: ${theme.palette.gray['0']};

        div {
            display: flex;
            flex-direction: column;
            justify-content: space-between;
            width: 100%;

            @media (min-width: ${theme.layout.breakpoints.md}) {
                flex-direction: row;
            }
        }
    `}
`;
