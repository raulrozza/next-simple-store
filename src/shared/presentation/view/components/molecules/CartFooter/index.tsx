import { FC } from 'react';

import { Button, Spacing } from '@/shared/presentation/view/components/atoms';

import { Footer } from './styles';

interface CartFooterProps {
    bottomLabels: string[];
    button: {
        text: string;
        onClick: () => void;
    };
}

const CartFooter: FC<CartFooterProps> = ({ bottomLabels, button }) => (
    <Footer>
        <div>
            {bottomLabels.map((label, index) => (
                <p key={`bottom-label-${index}`}>{label}</p>
            ))}
        </div>

        <Spacing size={2} />

        <Button variant="primary" onClick={button.onClick}>
            {button.text}
        </Button>
    </Footer>
);

export default CartFooter;
