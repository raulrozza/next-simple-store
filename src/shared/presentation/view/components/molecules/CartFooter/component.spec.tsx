import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { ThemeProvider } from '@/shared/presentation/contexts';

import CartFooter from './index';

describe('CartFooter', () => {
    it('should render correctly dispalying all the texts and the button', () => {
        const view = render(
            <CartFooter
                bottomLabels={['first label', 'second label']}
                button={{
                    text: 'Button',
                    onClick: jest.fn(),
                }}
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        expect(screen.getByText('first label')).toBeInTheDocument();
        expect(screen.getByText('second label')).toBeInTheDocument();
        expect(screen.getByText('Button')).toBeInTheDocument();
        expect(view).toMatchSnapshot();
    });

    it('should call the onClick function when the button is clicked', async () => {
        const user = userEvent.setup();

        const onClick = jest.fn();

        render(
            <CartFooter
                bottomLabels={[]}
                button={{
                    text: 'Button',
                    onClick,
                }}
            />,
            {
                wrapper: ThemeProvider,
            },
        );

        await user.click(screen.getByText('Button'));

        expect(onClick).toHaveBeenCalled();
    });
});
