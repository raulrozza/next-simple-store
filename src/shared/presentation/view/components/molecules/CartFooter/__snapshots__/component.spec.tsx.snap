// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartFooter should render correctly dispalying all the texts and the button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c2 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #579FB3;
  color: #FFFFFF;
}

.c2:hover {
  background-color: #71B7C8;
}

.c1 {
  margin-top: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media (min-width:768px) {
  .c0 div {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<body>
    <div>
      <footer
        class="c0"
      >
        <div>
          <p>
            first label
          </p>
          <p>
            second label
          </p>
        </div>
        <div
          class="c1"
          size="2"
        />
        <button
          class="c2"
        >
          Button
        </button>
      </footer>
    </div>
  </body>,
  "container": .c2 {
  padding: 0.5rem 1rem;
  border-radius: 0.5rem;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.25rem;
  background-color: #579FB3;
  color: #FFFFFF;
}

.c2:hover {
  background-color: #71B7C8;
}

.c1 {
  margin-top: 0.5rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  color: #FFFFFF;
}

.c0 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 100%;
}

@media (min-width:768px) {
  .c0 div {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div>
    <footer
      class="c0"
    >
      <div>
        <p>
          first label
        </p>
        <p>
          second label
        </p>
      </div>
      <div
        class="c1"
        size="2"
      />
      <button
        class="c2"
      >
        Button
      </button>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
